### PR TITLE
[jemalloc] Try jemalloc 5.3.0 (candidate)

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: J/jemalloc/build_tarballs.jl [hunk 1]
```diff
@@ -40,4 +54,4 @@ dependencies = Dependency[
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"11.0.0-iains")
```
Comment (giordano) [inline J/jemalloc/build_tarballs.jl:57]: Please, no :-1:

Comment (jeremiahpslewis) [reply J/jemalloc/build_tarballs.jl:57]: It's a draft, just wanted to see if it worked!

Comment (giordano) [reply J/jemalloc/build_tarballs.jl]: `v"11.0.0-iains"` doesn't even make sense for all platforms but `aarch64-apple-darwin` (where you use clang, not even gcc)

Comment (jeremiahpslewis) [reply J/jemalloc/build_tarballs.jl]: I was looking for a cheap way to trigger MacOS 11.0 and saw a code path which seemed to indicate that might work. But it doesn't resolve the build failure, so it's reverted. :)

**Pull request closed without merging**